### PR TITLE
Revert "Refactor HDF5 handlers and add DASK handler"

### PR DESCRIPTION
**Pull request by tacaswell**
Reverts NSLS-II/databroker#308

This breaks sub-classes which passing in 'key' as a string (as is still documented).

### CODE REVIEW

**Comment by danielballan** [comment]
Will merging this break eiger-io, @jrmlhermitte?

**Comment by danielballan** [comment]
I answered my own question: everything in eiger-io derives from `HandlerBase`, so this should be fine.

@stuwilkins Do you have the bandwidth to take another pass at #308 and do it in a backward-compatible way? It's OK if not; just let us know if we should pick this up.

**Comment by danielballan** [comment]
Randomly metadataserver failure, per usual. Merging!

File: databroker/assets/tests/test_handlers.py [hunk 1]
```diff
@@ -39,18 +37,12 @@ def fs_with_handlers(request, fs_cls):
     fs.register_handler('AD_HDF5_TS', AreaDetectorHDF5TimestampHandler)
     fs.register_handler('AD_HDF5_SWMR_TS',
                         AreaDetectorHDF5SWMRTimestampHandler)
-    fs.register_handler('AD_HDF5_DASK',
```
Comment (jrmlhermitte) [inline databroker/assets/tests/test_handlers.py:42]: this did not look like good practice here

Comment (jrmlhermitte) [reply databroker/assets/tests/test_handlers.py:42]: (Specs should be agnostic of file formats or how they're read right?)

**Pull request merged**